service/version_check.go

```diff
@@ -31,34 +31,36 @@ import (
 
 	"github.com/dchest/uniuri"
 
-	driver "github.com/arangodb/go-driver"
+	"github.com/arangodb/go-driver"
 
 	"github.com/arangodb-helper/arangodb/pkg/definitions"
 )
 
 // DatabaseVersion returns the version of the `arangod` binary that is being
 // used by this starter.
-
 func (s *Service) DatabaseVersion(ctx context.Context) (driver.Version, bool, error) {
-	for i := 0; i < 25; i++ {
-		d, enterprise, err := s.databaseVersion(ctx)
-		if err != nil {
-			s.log.Warn().Err(err).Msg("Error while getting version")
-			time.Sleep(time.Second)
-			continue
+	retries := 25
+	var err error
+	for i := 0; i < retries; i++ {
+		var v driver.Version
+		var enterprise bool
+		v, enterprise, err = s.databaseVersion(ctx)
+		if err == nil {
+			return v, enterprise, nil
 		}
 
-		return d, enterprise, nil
+		s.log.Warn().Err(err).Msgf("Error while getting version. Attempt %d of %d", i+1, retries)
```

Comment:
if ctx.Err() != nil {
// maybe SIGTERM was sent to the starter so this context probably is closed then, and `DatabaseVersion` should return ASAP
}

Reply:
The reason is known for long time - arangod by default uses background thread for writing the output. Main program can terminate before the background thread will flush the buffer. Added flag changes that.